data-es6.js

```diff
@@ -9649,7 +9649,9 @@ exports.tests = [
         var get = [];
         var p = new Proxy({}, { get: function(o, k) { get.push(k); return o[k]; }});
         Object.getOwnPropertyDescriptor(RegExp.prototype, 'flags').get.call(p);
-        return get + '' === "global,ignoreCase,multiline,unicode,sticky";
+        return get.indexOf('global') === 0 &&
+               get.indexOf('ignoreCase') === 1 &&
```

Comment:
My suggestion:
```js
var global = get.indexOf(‘global’);
var ignoreCase = get.indexOf(‘ignoreCase’);
var multiline = get.indexOf(‘multiline’);
var unicode = get.indexOf(‘unicode’);
var sticky = get.indexOf(‘sticky’);
return -1 < // this is weird indentation ofc, so indent however
  global && global <
  ignoreCase && ignoreCase <
  multiline && multiline <
  unicode && unicode <
  sticky;
```

Reply:
But why fail at all when `unicode` and `sticky` are absent?  That's not what's being tested here.

Reply:
If the engine supports those flags, and the proxy isn’t able to intercept them, then the proxy isn’t functioning correctly.

Reply:
The same could be said for `dotAll`, which your suggestion doesn't cover.  Let me put together a version of the test with feature detection.

Reply:
that's true, but this is the ES6 page, so it's fine if dotAll is missing (altho it'd be fine, if present, to ensure it's in the proper order)

Reply:
Okay, the new version checks whether each flag exists before expecting it to be hit by `#flags`.  Yeah, it's a bit silly to check for the ES3 flags, but it's makes the code more consistent and easier to modify if new flags are ever added.